src/components/EnhancedTable/EnhancedTable.jsx

```diff
@@ -46,153 +56,427 @@ class EnhancedTable extends React.Component {
+          style={style}
+          component="div"
+          variant="body"
+          padding="checkbox"
+          className={classNames(classes.tableCellFlex, colData.className)}
+        >
+          <Checkbox
+            checked={rowSelection.get('rowIsSelected')(rowData, rowKeyFn(rowData))}
+            onChange={onChangeHandler}
+          />
+        </TableCell>
+      );
+    }
+
+    const calcRowProps = this.getRowProps(rowData);
+
+    // ANY OTHER CELL
+    return (
+      <TableCell
+        {...calcRowProps}
+        style={style}
+        key={cellKey}
+        component="div"
+        variant="body"
+        className={classNames(classes.tableCellFlex, colData.className)}
+      >
+        {content}
+      </TableCell>
+    );
+  };
 
+  getColumnWidth = ({ index }) => {
+    const { columnMinWidth } = this.props;
     const {
-      columns,
-      dataSource,
-      rowKey,
-      rowProps,
-      className,
-      loading,
-      classes,
-      rowSelection,
-      fullHeight,
-      noDataText,
-      rowRenderOptions
-    } = this.props;
-
-    this.validateColumnKeys();
+      cols,
+      scrollbars,
+      width,
+      colsWithoutFixedWidth,
+      totalFixedWidth: fixedWidth,
+      numColsFixedWidth
+    } = this.state;
+
+    const remainingWidth = width - fixedWidth;
+    const colLength = cols.length;
+    const isVerticalScrollbarVisible = scrollbars && scrollbars.vertical;
+    const isLastColumn = index === colLength - 1;
+    const lastColWithoutFixedWidth = _.last(colsWithoutFixedWidth);
+    const isLastColWithoutFixedWidth = _.isEqual(lastColWithoutFixedWidth, cols[index]);
+    const unusedSpace = remainingWidth > 0 && numColsFixedWidth === colLength;
+    const numColsNoFixedWidth = colLength - numColsFixedWidth;
+
+    const padding =
+      isVerticalScrollbarVisible && (isLastColWithoutFixedWidth || (isLastColumn && unusedSpace)) ? scrollbars.size : 0;
+    if (cols[index].width) {
+      if (unusedSpace) {
+        return remainingWidth + cols[index].width - padding;
+      }
+      return cols[index].width;
+    }
+
+    if (remainingWidth <= 0) {
+      return columnMinWidth;
+    }
+
+    const adjustedWidth = numColsNoFixedWidth > 0 ? remainingWidth / numColsNoFixedWidth : columnMinWidth;
+
+    const finalWidth = adjustedWidth - padding < columnMinWidth ? columnMinWidth : adjustedWidth - padding;
+    return _.floor(finalWidth);
+  };
+
+  getTableHeight = () => {
+    const { rowHeight, height, maxHeight } = this.props;
+    const { data, headerData, scrollbars } = this.state;
+
+    if (height) {
+      return height;
+    }
+
+    if (!data) {
+      return `${rowHeight}px`;
+    }
+
+    const numVisibleRenderedRows = data.length === 0 ? 2 : data.length + headerData.length;
+
+    const padding = scrollbars && scrollbars.horizontal ? scrollbars.size : 0;
+    let calcHeight = rowHeight * numVisibleRenderedRows + padding;
+    if (calcHeight > maxHeight) {
+      calcHeight = maxHeight;
```

Comment:
Instead of using `let` could use `const` for `calcHeight` and just return `maxHeight` instead of re-assigning.